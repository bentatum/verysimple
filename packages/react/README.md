# @verysimple/react

This is the `@verysimple/react` package