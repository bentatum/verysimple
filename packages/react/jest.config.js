/** @type {import('jest').Config} */
module.exports = {
  testEnvironment: "jsdom",
  setupFilesAfterEnv: ["@testing-library/jest-dom/extend-expect"],
  collectCoverage: true,
  coverageThreshold: {
    global: {
      statements: 75,
      branches: 72,
      functions: 75,
      lines: 75,
    },
  },
  coverageReporters: ["text-summary", "json-summary"],
  collectCoverageFrom: ["src/**/*.{ts,tsx}", "!**/*.d.ts"],
  coverageDirectory: "coverage",
  moduleNameMapper: {
    "@/test/(.*)": "<rootDir>/test/$1",
    "@/(.*)$": "<rootDir>/src/$1",
  },
  modulePathIgnorePatterns: ["<rootDir>/dist/"],
  prettierPath: null,
};
