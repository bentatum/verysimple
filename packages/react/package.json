{
  "name": "@verysimple/react",
  "version": "8.1.1",
  "description": "A very simple component library",
  "main": "dist/index.js",
  "files": [
    "dist"
  ],
  "author": "Ben Tatum <mail@benjtatum.com>",
  "license": "MIT",
  "repository": "https://github.com/bentatum/verysimple",
  "scripts": {
    "test": "jest",
    "prebuild": "rm -rf dist",
    "build": "tsc --project tsconfig.build.json && tsc-alias"
  },
  "peerDependencies": {
    "react": ">=16",
    "react-dom": ">=16"
  },
  "devDependencies": {
    "@babel/core": "^7.20.12",
    "@babel/preset-env": "^7.19.1",
    "@babel/preset-react": "^7.18.6",
    "@babel/preset-typescript": "^7.18.6",
    "@testing-library/dom": "^8.20.0",
    "@testing-library/jest-dom": "^5.16.5",
    "@testing-library/react": "^13.4.0",
    "@testing-library/user-event": "^14.4.3",
    "@types/inquirer": "latest",
    "@types/jest": "^29.2.3",
    "@types/react": "^18.0.21",
    "@types/testing-library__jest-dom": "^5.14.5",
    "babel-jest": "^29.0.3",
    "babel-loader": "^8.2.5",
    "classnames": "^2.3.2",
    "framer-motion": "^10.15.0",
    "jest": "^29.0.3",
    "jest-environment-jsdom": "^29.0.3",
    "react": "^18.2.0",
    "react-dom": "^18.2.0",
    "tailwindcss": "^3.1.8",
    "tsc-alias": "^1.8.2",
    "typescript": "^4.9.3"
  },
  "volta": {
    "node": "20.10.0",
    "yarn": "1.22.21"
  },
  "dependencies": {
    "@headlessui/react": "^2.1.9",
    "@heroicons/react": "^2.0.13",
    "clsx": "^2.1.0",
    "framer-motion": "^10.15.0"
  }
}
