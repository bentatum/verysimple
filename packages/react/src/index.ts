export * from "./ButtonBase";
export * from "./Card";
export * from "./Dialog";
export * from "./FilledButtonBase";
export * from "./FilledIconButton";
export * from "./GradientBorder";
export * from "./helpers";
export * from "./IconButton";
export * from "./NormalIconButton";
export * from "./OutlinedButtonBase";
export * from "./OutlinedIconButton";
export * from "./Spinner";
export * from "./types";
