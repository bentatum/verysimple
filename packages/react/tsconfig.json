{
  "extends": "@verysimple/tsconfig/react-library.json",
  "compilerOptions": {
    "outDir": "./dist",
    "paths": {
      "@/*": ["./src/*"]
    }
  },
  "include": ["src/**/*.ts", "src/**/*.tsx"],
  "exclude": ["node_modules"]
}
