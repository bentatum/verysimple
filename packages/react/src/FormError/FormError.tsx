import classNames from "classnames";
import { ComponentPropsWithRef, forwardRef } from "react";

export interface FormErrorProps extends ComponentPropsWithRef<"div"> {}

export const FormError = forwardRef<HTMLDivElement, FormErrorProps>(
  (props, ref) => (
    <div
      {...props}
      ref={ref}
      role="alert"
      className={classNames(
        "mt-2 text-xs text-red-500 font-medium",
        props.className
      )}
    />
  )
);

FormError.displayName = "FormError";

export default FormError;
