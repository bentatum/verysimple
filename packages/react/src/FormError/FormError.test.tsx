import { render, screen } from "@testing-library/react";
import { createRef } from "react";
import FormError from "./FormError";

describe("FormError", () => {
  it("should be a red alert", () => {
    const { container } = render(<FormError>test</FormError>);
    expect(screen.getByRole("alert")).toHaveClass("text-red-500");
    expect(container).toMatchSnapshot();
  });

  it("forwards the ref", () => {
    const ref = createRef<HTMLParagraphElement>();
    render(<FormError ref={ref}>test</FormError>);
    expect(ref.current).toBeInTheDocument();
  });

  it("has a display name", () => {
    expect(FormError.displayName).toBe("FormError");
  });
});
