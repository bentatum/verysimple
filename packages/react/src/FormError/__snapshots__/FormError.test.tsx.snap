// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormError should be a red alert 1`] = `
<div>
  <div
    class="mt-2 text-xs text-red-500 font-medium"
    role="alert"
  >
    test
  </div>
</div>
`;
